packages/eslint-plugin-ecmascript-compat/lib/rule.js

```diff
@@ -1,21 +1,37 @@
 const compatibility = require('./compatibility');
 const { createDelegatee, delegatingVisitor } = require('./delegation');
 const features = require('./features');
+const polyfilling = require('./polyfilling');
 const targetRuntimes = require('./targetRuntimes');
 
 const targets = targetRuntimes();
-const delegateeConfigs = compatibility
-  .forbiddenFeatures(features, targets)
-  .map((feature) => feature.ruleConfig);
+const unsupportedFeatures = compatibility.unsupportedFeatures(features, targets);
 
 module.exports = {
   meta: {
     type: 'problem',
-    schema: [], // no options
+    schema: [
+      {
+        'type': 'object',
+        'properties': {
+          'polyfills': {
+            'type': 'array',
+            'items': {
+              'type': 'string'
+            }
+          }
+        },
+        'additionalProperties': 'false',
+      }
+    ]
   },
   create(context) {
-    const visitors = delegateeConfigs.map((config) => createDelegatee(config, context));
+      const polyfills = context.options?.[0]?.polyfills;
+      const validFeatures = unsupportedFeatures.filter(feature => !polyfilling.isPolyfilled(feature, polyfills));
```

Comment:
Apologies, I wrote a comment here before, but it looks like I didn't post it :(

I don't think the concept of "valid feature" is necessary, and we could do without some intermediate variables.

How about something like this to express the intent?

```
const forbiddenFeatures = unsupportedFeatures
  .filter(feature => !polyfilling.isPolyfilled(feature, polyfills))
const visitors = forbiddenFeatures .map((feature) => createDelegatee(feature.ruleConfig, context));
```

Reply:
Agreed. Should be much more readable. I've also got rid of polyfilling file and also isPolyfilled func (I think there's no much of readability gain in having a wrapper around simple array includes call after all).